src/reducers/query.jsx

```diff
@@ -187,6 +189,8 @@ function processParams( state, action ) {
     }
   } )
 
+  validatePer1000( processed )
```

Comment:
These could probably all move to above Line 778 (making the query string) since it is a calculated state

Reply:
You're right. I just tested it.  I had a hunch that I was putting the same line in so many places, but thought it was dependent only on a few actions.  I will change this.

Thank you